### PR TITLE
Refactor unit tests to use AEPTestUtils

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR refactors existing unit tests to use the AEPTestUtils and JSON comparison methods with the following changes:

**MediaPublicTrackerTests.swift**
1. All custom `isEqual` comparisons updated to use JSON comparison instead of custom dictionary casting and equality

**MediaXDMEventTests**
* testToXDMData - updated to use JSON comparison with strict equality
* For property `xdm.mediaCollection.sessionDetails.length` Is standard Int conversion fine (via AnyCodable decoder) instead of specific conversion to Int64?

**MediaXDMEventHelperTests**
1. Replace AssertUtils.compareSizeAndKeys with JSON comparison assertEqual

**XDMAdvertisingDetailsTests
XDMAdvertisingPodDetailsTests
XDMErrorDetailsTests
XDMChapterDetailsTests
XDMSessionDetailsTests
XDMMediaCollectionTests**
* `testEncode` does not have strict collection count equality validation so assertExactMatch was used - is exact equality desired instead?
    * Please note that as long as expected is valid, `actual` will be validated accordingly so guard/unwrap of `actual` is not strictly needed, unless there is a specific step that should be validated in depth

## Questions for reviewers
1. For dictionary value comparisons using the JSON comparison methods, `Double` values dont use the delta comparison logic present in `MediaPublicTrackerTests` `isEqual` method. Is this an acceptable change?:

from Sources -> Double+Media.swift:
```swift
// Adds functionality to compare the equality of two doubles to within the specified delta.
extension Double {
    func isAlmostEqual(_ doubleToCompare: Double, accuracy: Double = 0.000001) -> Bool {
        return fabs(self - doubleToCompare) < accuracy
    }
}
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Tests/UnitTests/MediaTests.swift [hunk 1]
```diff
@@ -13,6 +13,7 @@
 @testable import AEPCore
 @testable import AEPEdgeMedia
 import AEPServices
+import AEPTestUtils
```
Comment (emdobrin) [inline Tests/UnitTests/MediaTests.swift:16]: is this import used in this file?

File: Tests/UnitTests/Media+PublicAPITests.swift [hunk 2]
```diff
@@ -13,6 +13,7 @@
 @testable import AEPCore
 @testable import AEPEdgeMedia
 @testable import AEPServices
+import AEPTestUtils
 import XCTest
 
 class MediaPublicAPITests: XCTestCase {
```
Comment (emdobrin) [inline Tests/UnitTests/Media+PublicAPITests.swift:19]: this test is not updated yet, do you plan to update it in a future PR?

File: Tests/UnitTests/MediaPublicTrackerTests.swift [hunk 3]
```diff
@@ -147,7 +122,8 @@ class MediaPublicTrackerTests: XCTestCase {
         XCTAssertFalse((data?[MediaConstants.Tracker.ID] as? String ?? "").isEmpty)
 
         let actualParam = data?[MediaConstants.Tracker.EVENT_PARAM] as? [String: Any] ?? [:]
-        XCTAssertTrue(isEqual(map1: actualParam, map2: Self.testConfig))
+//        XCTAssertTrue(isEqual(map1: actualParam, map2: Self.testConfig))
```
Comment (emdobrin) [inline Tests/UnitTests/MediaPublicTrackerTests.swift:125]: remove commented code

File: Tests/UnitTests/MediaXDMEventTests.swift [hunk 4]
```diff
@@ -73,6 +75,31 @@ class MediaXDMEventTests: XCTestCase {
...
+                "playerName": "test_playerName",
+                "streamType": "video"
+              }
+            },
+            "timestamp": "\(Date(timeIntervalSince1970: 2).getISO8601UTCDateWithMilliseconds())"
+          }
+        }
+        """
+
+        assertEqual(expected: expected, actual: xdmEventData)
```
Comment (emdobrin) [inline Tests/UnitTests/MediaXDMEventTests.swift:102]: you may now remove the assertEqual statements above

**Review by kevinlind — COMMENTED** [review]
Are you able to remove the file `Tests/TestHelpers/TestUtils.swift` now as it appears you removed the use of the `flattenDictionary` function?

File: Tests/UnitTests/MediaTests.swift (hunk 1 above)
Comment (timkimadobe) [reply Tests/UnitTests/MediaTests.swift:16]: Yes, although it doesn't use the JSON assertion methods, it brings in the helpers:
1. `TestableExtensionRuntime`
2. `MockDataStore`

File: Tests/UnitTests/MediaPublicTrackerTests.swift (hunk 3 above)
Comment (timkimadobe) [reply Tests/UnitTests/MediaPublicTrackerTests.swift:125]: Removed!

File: Tests/UnitTests/Media+PublicAPITests.swift (hunk 2 above)
Comment (timkimadobe) [reply Tests/UnitTests/Media+PublicAPITests.swift:19]: Thank you, missed the assertions on this one - updated the dictionary property tests to use the JSON comparison methods

File: Tests/UnitTests/MediaXDMEventTests.swift (hunk 4 above)
Comment (timkimadobe) [reply Tests/UnitTests/MediaXDMEventTests.swift:102]: Thank you! Removed the refactored assertions

**Review by timkimadobe — COMMENTED** [review]
Thanks for the reviews @emdobrin and @kevinlind! Updated based on feedback

I didn't remove the `TestUtils.swift` dictionary flattening helpers because they're still used by the functional and integration tests - I will remove this helper in the PR that updates those tests to use AEPTestUtils

Also moving this PR to a new branch to clean up commit history with the recent releases: #55

**Pull request closed without merging**